### DISCUSSION TITLE
How do I create partitions for only business days?

### DISCUSSION THREAD

**Discussion by clairelin135** [dagster-io/dagster#14528] [Q&A] [answered]
`DailyPartitionsDefinitions` select every day as a partition. How do I select only the business days?

**Comment by clairelin135** [comment, marked as answer]
There are a couple options for creating partitions on business days:
1. Create a `TimeWindowPartitionsDefinition` with a cron schedule of `0 0 * * 1-5`, which creates a partition for Monday through Friday at 00:00.
2. The approach above works if all Monday through Friday days are valid. But if you want to take holidays into account too, you can define a `DynamicPartitionsDefinition` (currently experimental) and create a partition for each business day. For example:
```
business_days = DynamicPartitionsDefinition(name="business_days")


@op
def add_business_day_partition(context):
    yesterday = date.today() - timedelta(days=1)
    if is_business_day(yesterday):
        context.instance.add_dynamic_partitions(business_days.name, [yesterday.strftime("%Y-%m-%d")])


@job(partitions_def=DailyPartitionsDefinition("2023-01-01"))
def add_business_day_partition_job():
    add_business_day_partition()

defs = Definitions(
    schedules=[
        build_schedule_from_partitioned_job(add_business_day_partition_job),
    ],
)

```

**Comment by rprechelt** [comment]
@clairelin135 When using your `DynamicPartitionsDefinition` method above, the partition shows up as a "StaticPartition" in the DagsterUI when viewing partitions and materializing assets, i.e. you lose the "date/time selector" which has useful "drag and select" in the DagsterUI. 

Is it possible to use a unique set of datetime periods (e.g. only business days, or only certain hours on certain) that the user provides but preserve the "date/time selector" in the DagsterUI?
